lib/client/api/sharing.js

```diff
@@ -175,17 +189,140 @@ module.exports = function (jiffClient) {
+   *                                   will get a unique id based on the concatenation of base_share_id and the index of the element.
+   *                                   This tag is used so that parties distinguish messages belonging to this share operation from
+   *                                   other share operations between the same parties (when the order of execution is not
+   *                                   deterministic). An automatic id is generated by increasing a local counter, default
+   *                                   ids suffice when all parties execute all sharing operations with the same senders
+   *                                   and receivers in the same order.
+   * @returns {promise} if the calling party is a receiver then a promise to the shared arrays is returned, the promise will provide an object
+   *                    formatted as follows: { <party_id>: [ [ <1st_row_shares> ], [<2nd_row_share> ], ..., [ <(lengths[party_id])th_row_shares> ] ] }
+   *                    where the party_ids are those of the senders.
+   *                    if the calling party is not a receiver, then null is returned.
+   */
+  jiffClient.share_2D_array = arraysSharing.jiff_share_2D_array.bind(null, jiffClient);
+
+  /**
+   * Share an n-dimensional array of secrets
+   * The arrays can be of different lengths and dimensions.
+   * @method share_ND_array
+   * @memberof module:jiff-client~JIFFClient
+   * @instance
+   * @returns {object|promise}
+   */
+  jiffClient.share_ND_array = arraysSharing.jiff_share_ND_array.bind(null, jiffClient);
+
+  /**
+   * Helper function of share_ND_array
+   * This method gets called if share_ND_array does not have enough information
+   *                 to output an array of shares immediately.  Instead it
+   *                 returns a promise to an object holding the unknown-size
+   *                 arrays from each sender.
+   * @method share_ND_array_deferred
+   * @memberof module:jiff-client~JIFFClient
+   * @instance
+   * @returns {promise}
+   */
+  jiffClient.share_ND_array_deferred = arraysSharing.jiff_share_ND_array_deferred.bind(null, jiffClient);
+
+  /**
+   * Helper function of share_ND_array
+   * This method gets called if share_ND_array when share_ND_array was given
+   *                 sufficients information (in the form of array skeletons) to
+   *                 infer the size and shape of each of the senders' arrays.
+   *                 This allows it to immediately return an object containing
+   *                 the secret-shared arrays from each sender.
+   * @method share_ND_array_static
+   * @memberof module:jiff-client~JIFFClient
+   * @instance
+   * @returns {object}
+   */
+  jiffClient.share_ND_array_static = arraysSharing.jiff_share_ND_array_static.bind(null, jiffClient);
+
+  /*
+   * Wipe a secret array of all secrets but preserve the shape
+   */
+  jiffClient.skeleton_of = arraysSharing.jiff_skeleton_of.bind(null, jiffClient);
```

Comment:
this we need to expose because it can be useful, but please add a reference to the docs of share_ND_array and all its alias (1D and 2D arrays) that refer to this using @see and says how this can be used to provide a skeleton.